internal/services/cdn/cdn_frontdoor_route_disable_link_to_default_domain_resource.go

```diff
@@ -72,63 +71,63 @@ func resourceCdnFrontDoorRouteDisableLinkToDefaultDomainCreate(d *pluginsdk.Reso
 
 	customDomains := d.Get("cdn_frontdoor_custom_domain_ids").([]interface{})
 
-	routeId, err := parse.FrontDoorRouteID(d.Get("cdn_frontdoor_route_id").(string))
+	routeId, err := parse.FrontDoorRouteIDInsensitively(d.Get("cdn_frontdoor_route_id").(string))
 	if err != nil {
-		return err
+		return fmt.Errorf("creating Front Door Route Disable Link To Default Domain: %+v", err)
 	}
 
 	// create the resource id
 	uuid, err := uuid.GenerateUUID()
 	if err != nil {
 		return fmt.Errorf("generating UUID: %+v", err)
 	}
+
 	id := parse.NewFrontDoorRouteDisableLinkToDefaultDomainID(routeId.SubscriptionId, routeId.ResourceGroup, routeId.ProfileName, routeId.AfdEndpointName, routeId.RouteName, uuid)
 
-	// Lock the resources to make sure this will be an atomic operation
-	// as this is a one to many association potentially(e.g. route -> custom domain(s))...
 	locks.ByName(routeId.RouteName, cdnFrontDoorRouteResourceName)
 	defer locks.UnlockByName(routeId.RouteName, cdnFrontDoorRouteResourceName)
 
-	// Lock all of the custom domains associated with this unlink default domain resource
-	// prolly overkill but we don't want the custom domains moving while we are doing this
-	// operation as well...
 	for _, v := range customDomains {
-		customDomainId, err := parse.FrontDoorCustomDomainID(v.(string))
+		customDomainId, err := parse.FrontDoorCustomDomainIDInsensitively(v.(string))
```

Comment:
this is coming from the config which is validated and so should be being parsed case-sensitively:

```suggestion
		customDomainId, err := parse.FrontDoorCustomDomainID(v.(string))
```

Reply:
Fixed.